code/game/objects/items/weapons/storage/storage.dm

```diff
@@ -32,7 +36,6 @@
 			return
 
 		if(over_object == M && Adjacent(M)) // this must come before the screen objects only block
-			orient2hud(M)          // dunno why it wasn't before

```

Comment:
This orient2hud (and several others) are moved into show_to, since that is what ultimately needs it.